index/scorch/persister.go

```diff
@@ -268,13 +290,21 @@ func (s *Scorch) parsePersisterOptions() (*persisterOptions, error) {
 	return &po, nil
 }
 
-func (s *Scorch) persistSnapshot(snapshot *IndexSnapshot) error {
-	persisted, err := s.persistSnapshotMaybeMerge(snapshot)
-	if err != nil {
-		return err
-	}
-	if persisted {
-		return nil
+func (s *Scorch) persistSnapshot(snapshot *IndexSnapshot,
+	po *persisterOptions) error {
+	// With ActiveMergeMemorySegments disabled, perform in-memory segment
+	// merging only when the memory pressure is below the configured,
+	// threshold else persister always performs the memory segments merge
+	// before persistence.
+	if po.ActiveMergeMemorySegments ||
+		s.paused() <= uint64(po.MemoryPressurePauseThreshold) {
```

Comment:
two thoughts here, where the 2nd one might obsolete the first one...

A) since these two options are so "close together" in terms of code location and the area that they're meant to control, a better naming approach is if they somehow shared the same naming prefix.  Instead of "xxYyZzFoo" and "barYyZzXx", might as well aim for a better name.

B) But, instead of trying for a better name, perhaps instead reduce from two options down to a single option.  To my eyes, if you change the logic to...

  if paused() < uint64(po.MemoryPressurePauseThreshold) { ... do-something ... }

Note the change from <= to just <, so by setting MemoryPressurePauseThreshold to zero, it would be the same as saying never go into the "do-something" codepath.   Similarly, cranking up MemoryPressurePauseThreshold to a super high number means always go into the "do-something" codepath.

And, that would have the benefit of one less knob to think about and explain to folks.

Reply:
Indeed!